cpp/oneapi/dal/backend/primitives/selection/test/select_flagged_overflow_dpc.cpp

```diff
@@ -0,0 +1,140 @@
+#include "oneapi/dal/test/engine/math.hpp"
+#include "oneapi/dal/backend/primitives/selection/select_flagged.hpp"
+
+namespace oneapi::dal::backend::primitives::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+namespace de = dal::detail;
+
+template <typename TestType>
+class select_flagged_overflow_test : public te::policy_fixture {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Flag = std::tuple_element_t<1, TestType>;
+
+    auto allocate_dummy_arrays(std::int64_t dummy_elem_count) {
+        auto policy = de::data_parallel_policy{ this->get_queue() };
+        Float* val_ptr = de::data_parallel_allocator<Float>{ policy }.allocate(elem_count_);
+        Flag* mask_ptr = detail::data_parallel_allocator<Flag>{ policy }.allocate(elem_count_);
+
+        auto val = ndarray<Float, 1>::wrap(val_ptr,
+                                           { dummy_elem_count },
+                                           detail::make_default_delete<Float>(policy));
+        auto mask = ndarray<Flag, 1>::wrap(mask_ptr,
+                                           { dummy_elem_count },
+                                           detail::make_default_delete<Flag>(policy));
+
+        return std::make_tuple(val, mask);
+    }
+
+    void check_select(ndarray<Float, 1>& in, ndarray<Flag, 1>& mask) {
+        auto policy = de::data_parallel_policy{ this->get_queue() };
+        Float* out_ptr = de::data_parallel_allocator<Float>{ policy }.allocate(elem_count_);
+
+        auto out = ndarray<Float, 1>::wrap(out_ptr,
+                                           { in.get_count() },
+                                           detail::make_default_delete<Float>(policy));
+        std::int64_t total_sum_res;
+
+        select_flagged<Float, Flag>{ this->get_queue() }(mask, in, out, total_sum_res)
+            .wait_and_throw();
+    }
+
+    std::int64_t get_value_to_overflow() {
+        return 0x7FFFFFFFFFFFFFFF;
+    }
+
+private:
+    const std::int64_t elem_count_ = 1;
+};
+
+template <typename TestType>
+class select_flagged_index_overflow_test : public te::policy_fixture {
+public:
+    using Data = std::tuple_element_t<0, TestType>;
+    using Flag = std::tuple_element_t<1, TestType>;
+
+    auto allocate_dummy_arrays(std::int64_t dummy_elem_count) {
+        auto policy = de::data_parallel_policy{ this->get_queue() };
+        Data* val_ptr = de::data_parallel_allocator<Data>{ policy }.allocate(elem_count_);
+        Flag* mask_ptr = detail::data_parallel_allocator<Flag>{ policy }.allocate(elem_count_);
+
+        auto val = ndarray<Data, 1>::wrap(val_ptr,
+                                          { dummy_elem_count },
+                                          detail::make_default_delete<Data>(policy));
+        auto mask = ndarray<Flag, 1>::wrap(mask_ptr,
+                                           { dummy_elem_count },
+                                           detail::make_default_delete<Flag>(policy));
```

Comment:
Use `ndarray::empty(q, { dummy_elem_count }, sycl::usm::alloc::device)`

Reply:
This block for creating dummy ndarray for testing overflow checks, therefore  i can't use ndarray::empty here.